src/org/skypro/skyshop/basket/ProductBasket.java

```diff
@@ -31,12 +38,14 @@ public int sumProductBasket() {
     //Метод для подсчета количества специальных товаров
     public int countSpecialProduct() {
         int count = 0;
-        for (Product product : basket) {
-            if (product != null) {
-                if (product.isSpecial() == true) {
-                    count = count + 1;
-                }
+        for (LinkedList<Product> basket : basket.values()) {
```

Comment:
ссылки должны быть всегда интерфейсные, чтобы соблюдать полиморфизм

```suggestion
        for (List<Product> basket : basket.values()) {
```